src/view/observer/mutationobserver.js

```diff
@@ -240,16 +240,16 @@ export default class MutationObserver extends Observer {
 			// Anchor and focus has to be properly mapped to view.
 			if ( viewSelectionAnchor && viewSelectionFocus ) {
 				viewSelection = new ViewSelection();
-				viewSelection.setTo( viewSelectionAnchor );
-				viewSelection.setFocus( viewSelectionFocus );
+				viewSelection._setTo( viewSelectionAnchor );
```

Comment:
We need a separate `Selection` class with the public API and `DocumentSelection` used by document, simillary to what we have in the model.

Reply:
Same for all other observers which uses protected methods.